freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/EnvironmentAccessChecker.java

```diff
@@ -0,0 +1,82 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.cloudera.thunderhead.service.authorization.AuthorizationProto;
+import com.sequenceiq.authorization.resource.AuthorizationResource;
+import com.sequenceiq.authorization.resource.ResourceAction;
+import com.sequenceiq.authorization.resource.RightUtils;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnParseException;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.cloudbreak.auth.security.InternalCrnBuilder;
+import com.sequenceiq.freeipa.service.freeipa.user.model.EnvironmentAccessRights;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+
+public class EnvironmentAccessChecker {
```

Comment:
I would annotate this class with `@Service` and inject it into the proper class or make it static (maybe `@Service` is better because you can write mockito tests for it), instead of instantiate it every time

Reply:
I instantiate this each time I sync an environment so I can re-use the resource-dependent right check "environments/accessEnvironment".
If you prefer, I can try to swap this out for a Prototype bean and BeanFactory.

Reply:
If @keyki or @lacikaaa thinks it is ok, then you can leave this as-is, but you can reuse the result of `hasAccess` anywhere as a local variable or parameter

Reply:
I'm not re-using the result because I'm calling hasAccess only once for each user and machineUser.

Reply:
this should be ok